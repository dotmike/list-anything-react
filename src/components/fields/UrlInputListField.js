import React from "react";
import PropTypes from "prop-types";
import { Form } from "semantic-ui-react";

import ListValues from "./ListValues";
import EditableListInput from "./EditableListInput";

const styles = {
  showFieldDescription: {
    marginTop: "0px",
    maxHeight: "100px",
    overflowY: "auto"
  },
  default: { maxHeight: "100px", overflowY: "auto" }
};

class UrlInputListField extends React.Component {
  constructor(props) {
    super(props);
    this.values = props.value ? props.value.split("\n") : [];
  }

  shouldComponentUpdate = nextProps =>
    this.props.editable !== nextProps.editable ||
    this.props.showFieldDescription !== nextProps.showFieldDescription ||
    this.props.error !== nextProps.error;

  onChange = e => {
    this.values = e.target.values;
    this.props.onChange(e);
  };

  render() {
    const { editable, showFieldDescription, error, field } = this.props;
    const { values } = this;

    if (editable)
      return (
        <EditableListInput
          values={values}
          type="url"
          error={error}
          field={field}
          onChange={this.onChange}
        />
      );
    if (showFieldDescription)
      return (
        <Form.Field>
          <b>{field.description}:</b>
          <ListValues values={values} style={styles.showFieldDescription} />
        </Form.Field>
      );
    return <ListValues values={values} style={styles.default} />;
  }
}

UrlInputListField.propTypes = {
  // ownProps
  value: PropTypes.string.isRequired,
  editable: PropTypes.bool,
  showFieldDescription: PropTypes.bool,
  error: PropTypes.string,
  field: PropTypes.shape({
    _id: PropTypes.string,
    description: PropTypes.string
  }),
  onChange: PropTypes.func
};

UrlInputListField.defaultProps = {
  // ownProps
  editable: false,
  showFieldDescription: true,
  error: "",
  field: {},
  onChange: () => {}
};

export default UrlInputListField;
