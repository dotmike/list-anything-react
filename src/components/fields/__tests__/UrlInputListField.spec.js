import React from "react";
import { shallow } from "enzyme";

import UrlInputListField from "../UrlInputListField";

const defaultProps = {
  value: "testando...",
  error: "",
  editable: true,
  showFieldDescription: false,
  field: { _id: "1", description: "Testando:" },
  onChange: () => {}
};

describe("UrlInputListField", () => {
  describe("when `editable` is true", () => {
    const props = { ...defaultProps, editable: true };

    it("renders correctly", () => {
      const wrapper = shallow(<UrlInputListField {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when `editable` is false", () => {
    let props;

    beforeEach(() => {
      props = { ...defaultProps, editable: false };
    });

    it("renders correctly when `showFieldDescription` is false", () => {
      props.showFieldDescription = false;

      const wrapper = shallow(<UrlInputListField {...props} />);
      expect(wrapper).toMatchSnapshot();
    });

    it("renders correctly when `showFieldDescription` is true", () => {
      props.showFieldDescription = true;

      const wrapper = shallow(<UrlInputListField {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
