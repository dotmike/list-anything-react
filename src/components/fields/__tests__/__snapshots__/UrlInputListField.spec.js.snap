// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlInputListField when \`editable\` is false renders correctly when \`showFieldDescription\` is false 1`] = `
<ListField
  onRemove={null}
  style={
    Object {
      "maxHeight": "100px",
      "overflowY": "auto",
    }
  }
  values={
    Array [
      "testando...",
    ]
  }
/>
`;

exports[`UrlInputListField when \`editable\` is false renders correctly when \`showFieldDescription\` is true 1`] = `
<FormField>
  <b>
    Testando:
    :
  </b>
  <ListField
    onRemove={null}
    style={
      Object {
        "marginTop": "0px",
        "maxHeight": "100px",
        "overflowY": "auto",
      }
    }
    values={
      Array [
        "testando...",
      ]
    }
  />
</FormField>
`;

exports[`UrlInputListField when \`editable\` is true renders correctly 1`] = `
<EditableListInput
  error=""
  field={
    Object {
      "_id": "1",
      "description": "Testando:",
    }
  }
  onChange={[Function]}
  type="url"
  values={
    Array [
      "testando...",
    ]
  }
/>
`;
